orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/RestartStageHandler.kt

```diff
@@ -58,13 +59,38 @@ class RestartStageHandler(
         if (topStage.status.isComplete) {
           topStage.addRestartDetails(message.user)
           topStage.reset()
+          restartParentPipelineIfNeeded(message, topStage)
           repository.updateStatus(topStage.execution.type, topStage.execution.id, RUNNING)
           queue.push(StartStage(startMessage))
         }
       }
     }
   }
 
+  private fun restartParentPipelineIfNeeded(message: RestartStage, topStage: Stage) {
+    val trigger = topStage.execution.trigger
+    if (!topStage.execution.trigger.type.equals("pipeline")) {
+      return
+    }
+
+    val parentExecution = trigger.other["parentExecution"] as Execution
+    val parentStage = parentExecution.stageById(trigger.other["parentPipelineStageId"] as String)
+    parentStage.addAlreadyRunning(message)
+
+    queue.push(RestartStage(parentExecution, parentStage.id, message.user))
+  }
+
+  /**
+   * Inform the parent stage when it restarts that the child is already running
+   */
+  private fun Stage.addAlreadyRunning(message: RestartStage) {
+    message.withExecution { execution ->
```

Comment:
this seems weird, why add these props on the execution it should be on the stage... but also... let me think about a better way of doing this

Reply:
ok, so i think it would be better to do the following:
1. add something to the stage context like `context["_skipPipelineRestart"] = true`
2. No need to add `executionId` or `executionName` since those are already there
3. You will need to store the execution to persist this change to the stage in the executionRepository
4. `PipelineStage.prepareStageForRestart` should check the `_skipPipelineRestart` field, and if `true` not delete the `executionId/executionName`, from the context. It should *also clear* the `_skipPipelineRestart` flag so that if the user wanted to actually restart it explicitly afterwards it would work as expected (e.g. restart the child)
5. the `PipelineStage.taskGraph` should only create the start pipelinetask if there is no `executionId` present in the *current* stage context.

Reply:
Alright I have addressed your points, thanks for the feedback!

I did not really understand point 3 though. I think this is already being done elsewhere in the code, but I might be wrong.

Also, regarding point 5, the stage context passed to `taskGraph` is the *current* one, right?

Let me know what you think, and thanks again!